robyn/__init__.py

```diff
@@ -204,99 +205,99 @@ def inner(handler):
 
         :param endpoint str: endpoint to server the route
         """
 
         def inner(handler):
-            return self._add_route("PUT", endpoint, handler)
+            return self._add_route("PUT", endpoint, handler, validate_params=validate)
 
         return inner
 
-    def delete(self, endpoint: str):
+    def delete(self, endpoint: str, validate: bool = False):
         """
         The @app.delete decorator to add a get route
 
         :param endpoint str: endpoint to server the route
         """
 
         def inner(handler):
-            return self._add_route("DELETE", endpoint, handler)
+            return self._add_route("DELETE", endpoint, handler, validate_params=validate)
 
         return inner
 
-    def patch(self, endpoint: str):
+    def patch(self, endpoint: str, validate: bool = False):
         """
         [The @app.patch decorator to add a get route]
 
         :param endpoint [str]: [endpoint to server the route]
         """
 
         def inner(handler):
-            return self._add_route("PATCH", endpoint, handler)
+            return self._add_route("PATCH", endpoint, handler, validate_params=validate)
 
         return inner
 
-    def head(self, endpoint: str):
+    def head(self, endpoint: str, validate: bool = False):
         """
         The @app.head decorator to add a get route
 
         :param endpoint str: endpoint to server the route
         """
 
         def inner(handler):
-            return self._add_route("HEAD", endpoint, handler)
+            return self._add_route("HEAD", endpoint, handler, validate_params=validate)
 
         return inner
 
-    def options(self, endpoint: str):
+    def options(self, endpoint: str, validate: bool = False):
         """
         The @app.options decorator to add a get route
 
         :param endpoint str: endpoint to server the route
         """
 
         def inner(handler):
-            return self._add_route("OPTIONS", endpoint, handler)
+            return self._add_route("OPTIONS", endpoint, handler, validate_params=validate)
 
         return inner
 
-    def connect(self, endpoint: str):
+    def connect(self, endpoint: str, validate: bool = False):
         """
         The @app.connect decorator to add a get route
 
         :param endpoint str: endpoint to server the route
         """
 
         def inner(handler):
-            return self._add_route("CONNECT", endpoint, handler)
+            return self._add_route("CONNECT", endpoint, handler, validate_params=validate)
 
         return inner
 
-    def trace(self, endpoint: str):
+    def trace(self, endpoint: str, validate: bool = False):
         """
         The @app.trace decorator to add a get route
 
         :param endpoint str: endpoint to server the route
         """
 
         def inner(handler):
-            return self._add_route("TRACE", endpoint, handler)
+            return self._add_route("TRACE", endpoint, handler, validate_params=validate)
```

Comment:
Also, is there a reason for passing `validate_params` as kwargs? And we can also name the variables in the outer and the inner functions for more consistency.

Reply:
I used kwargs because is_const comes before validate_params in _add_route (at least I kept it like that). Perhaps I didn't need to do that?

Reply:
@kliwongan , we can change that. :)

>  is_const comes before validate_params in _add_route (at least I kept it like that). Perhaps I didn't need to do that?